spec/models/api_request_spec.rb

```diff
@@ -0,0 +1,53 @@
+require "rails_helper"
+require "fake_issues"
+
+RSpec.describe ApiRequest do
+  describe "#response" do
+    describe "with valid host, path, and query" do
+      it "returns an object with a code and a body" do
+        # stub the net/http request / response
+        success = Net::HTTPSuccess.new(1.0, "200", "OK")
+        allow_any_instance_of(ApiRequest)
+          .to receive(:send_api_request)
+          .and_return success
+        expect(success).to receive(:body).and_return('{"meta":{},"data":[]}')
+
+        # mock in host, path, and query
+        allow_any_instance_of(ApiRequest).to receive(:host).and_return "example.com"
+        allow_any_instance_of(ApiRequest).to receive(:path).and_return "/path"
+        allow_any_instance_of(ApiRequest).to receive(:query).and_return "?some=query"
+
+        res = ApiRequest.new.response
+        expect(res).to match({code: "200", body: {"meta" => {}, "data" => []}})
+      end
+
+      it "is memoized" do
+        api = ApiRequest.new
+        expect(api).to receive(:get_response).once.and_return "value"
+        api.response
```

Comment:
[thought] i think i would prefer this to be written as 
`expect(api.response).to eq api.response` since that is an expectation as well as calling the method twice. (but this is more of a nitpick i guess? not blocking for sure)